core/src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -87,9 +83,8 @@ public GatewayAllocator(Settings settings, TransportNodesListGatewayStartedShard
         logger.debug("using initial_shards [{}]", initialShards);
     }
 
-    public void setReallocation(final ClusterService clusterService, final AllocationService allocationService) {
-        this.clusterService = clusterService;
-        this.allocationService = allocationService;
+    public void setReallocation(final ClusterService clusterService, final RoutingService routingService) {
```

Comment:
oh man that class is a nighmare. really I just realized how fucked up this is grrr.